### PR TITLE
Fix52

### PR DESCRIPTION
**Pull request by mattcollier**
@dlongley failure due to race condition with this patch is:
- local: ~1/30
- aws/jenkins: ~1/60

### CODE REVIEW

File: lib/election.js [hunk 1]
```diff
@@ -941,9 +993,28 @@ function _getEventFromPeer(ledgerNode, peerId, eventHash, callback) {
   }, callback);
 }
 
+function _getVoteFromPeer(
+  ledgerNodeId, peerId, voteHash, electionTopic, callback) {
+  async.auto({
+    vote: callback => api._client.getVote(voteHash, peerId, callback),
+    hash: ['vote', (results, callback) => api._hasher(results.vote, callback)],
+    store: ['hash', (results, callback) => api._storage.votes.add(
```
Comment (mattcollier) [inline lib/election.js]: @dlongley I believe you commented that we need to do more validation here?

File: lib/election.js [hunk 2]
```diff
@@ -941,9 +993,28 @@ function _getEventFromPeer(ledgerNode, peerId, eventHash, callback) {
...
+        }
+        callback(err);
+      })]
+  }, callback);
+}
+
 function _createManifest(ledgerNodeId, blockHeight, type, items, callback) {
   // create manifest hash (EOL delimited) and store manifest
-  const hash = niUri.digest('sha-256', items.join('\n'), true);
+  const hash = niUri.digest('sha-256', items.sort().join('\n'), true);
```
Comment (mattcollier) [inline lib/election.js]: this should use `api.createManifestHash`

File: lib/voteStorage.js [hunk 3]
```diff
@@ -150,6 +150,24 @@ api.exists = (ledgerNodeId, voteHash, callback) => {
   });
 };
 
+// TODO: add documentation
```
Comment (mattcollier) [inline lib/voteStorage.js]: Add documentation.

File: lib/voteStorage.js [hunk 4]
```diff
@@ -175,6 +193,18 @@ api.get = (ledgerNodeId, blockHeight, electionTopic, callback) => {
...
```
Comment (mattcollier) [inline lib/voteStorage.js]: Add documentation.

File: lib/election.js [hunk 5]
```diff
@@ -1032,9 +1031,8 @@ function _createManifest(ledgerNodeId, blockHeight, type, items, callback) {
 }
 
 function _createVote(ledgerNode, electionTopic, vote, voter, callback) {
-  // TODO: add logger.verbose() with the vote information here
-  // console.log(voter.id.split('/').pop() + ' votes for ' + electionTopic +
-  //   ', manifestHash=' + vote.manifestHash + ', round=' + vote.voteRound);
+  logger.verbose(voter.id.split('/').pop() + ' votes for ' + electionTopic +
```
Comment (dlongley) [inline lib/election.js]: @mattcollier,

This `voter.id.split('/').pop()` stuff probably needs to change to just `voter.id` (shouldn't assume too much about how these look) ... and we should use the meta data field with the logger and move and/or duplicate some of these items into logged meta data properties. I don't remember how that works, can you take a stab?

Comment (mattcollier) [reply lib/election.js]: @dlongley the api is (message, meta) where meta must be an object.  What do you want the logger message to say?

**Pull request merged**